src/cdk/testing/testbed/testbed-harness-environment.ts

```diff
@@ -56,6 +64,24 @@ export class TestbedHarnessEnvironment extends HarnessEnvironment<Element> {
     await this._fixture.whenStable();
   }
 
+  async waitForAsyncTasksToComplete(): Promise<void> {
```

Comment:
I'd prefer a name that makes it more clear this is waiting for tasks outside of `NgZone`, maybe `waitForTasksOutsideAngular` since it corresponds to `runOutsideAngular`

Reply:
I had that initially, but I felt like it is confusing to mention Angular here at all. The logic we have in place basically waits for _all_ async tasks, unlike `whenStable` which is a Angular-specific NgZone method.

If we name it `outsideAngular`, one might think it _only_ waits for the tasks outside of the Angular zone.. which is obviously not the case.

Reply:
I like having a name that lines up with `runOutsideAngular` because it makes it more clear when this method should be called. Yes, it also waits for tasks inside `NgZone`, but you don't need to call it for that, that happens automatically. You need to call this if you have `runOutsideAngular` in your component code and need to clear that task

Reply:
Yeah, I see that point. It just semantically doesn't match what the function does. Maybe we can make that more clear but somehow still mention `outsideAngular`? I don't know of a good name for that though.

Reply:
`waitForTasksInTheParentZoneOfNgZone`? 😃

Reply:
I don't feel strong about either option. It's just that I feel like it's not ideal to make the function name less correct in favor of making it easier to find. The function basically has side-effects then.

Happy to just go with `waitForTasksOutsideAngular` or explore more options 😄

Reply:
I also like `waitForTasksOutsideAngular`. It's the most direct connection to what the component itself would be doing

Reply:
I'll update it. My feeling is still that method names should describe what is done semantically, rather than serving as an example of when to use the method. Though I don't know any better name that is correct plus mentions `outsideAngular`, so 🤷‍♂

Reply:
Done.